### PR TITLE
Add setSecret to mock-task.

### PR DESCRIPTION
**Pull request by hashtagchris**
Add a mock for `setSecret`. Currently unit tests fail if you try to use `setSecret` in your task.

### CODE REVIEW

File: node/package-lock.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "azure-pipelines-task-lib",
-  "version": "3.0.0-preview",
+  "version": "3.0.1-preview",
   "lockfileVersion": 1,
```
Comment (hashtagchris) [inline node/package-lock.json]: FYI, I ran `npm install npm && npm install` and there were a lot more changes in this file. All the sliding dependencies like `^2.2.2` were locked onto a particular version.

Is this due to using the latest npm? Or is there some npm configuration we should be using?

File: node/package.json [hunk 2]
```diff
@@ -27,10 +27,10 @@
   "dependencies": {
     "minimatch": "3.0.4",
     "mockery": "^1.7.0",
-    "uuid": "^3.0.1",
     "q": "^1.1.2",
     "semver": "^5.1.0",
-    "shelljs": "^0.3.0"
+    "shelljs": "^0.3.0",
+    "uuid": "^3.0.1"
```
Comment (hashtagchris) [inline node/package.json]: npm alphabetized the list of dependencies.

**Comment by bryanmacfarlane** [comment]
I would default bug level fixes to 2.x

**Comment by hashtagchris** [comment]
This isn't urgent for me, yet. It's unclear how quickly we'll get our tasks bumped up to consume ^2.8.0. I'll prep the cherry-pick. If you're good with the change, please merge.

**Pull request merged**